src/kuzzleSearchResult.js

```diff
@@ -0,0 +1,170 @@
+var KuzzleDocument = require('./kuzzleDocument.js');
+
+/**
+ *
+ * @param {KuzzleDataCollection} dataCollection
+ * @param {int} total
+ * @param {KuzzleDocument[]} documents
+ * @param {object} [aggregations]
+ * @param {object} [searchArgs]
+ * @param {KuzzleSearchResult} [previous]
+ * @constructor
+ */
+function KuzzleSearchResult (dataCollection, total, documents, aggregations, searchArgs, previous) {
+  Object.defineProperties(this, {
+    // read-only properties
+    dataCollection: {
+      value: dataCollection,
+      enumerable: true
+    },
+    total: {
+      value: total,
+      enumerable: true
+    },
+    documents: {
+      value: documents,
+      enumerable: true
+    },
+    aggregations: {
+      value: aggregations || {},
+      enumerable: true
+    },
+    searchArgs: {
+      value: searchArgs || {},
+      enumerable: true
+    },
+    // writable properties
+    _previous: {
+      value: previous || null,
+      enumerable: true,
+      writable: true
+    },
+    _next: {
+      value: null,
+      enumerable: true,
+      writable: true
+    }
+  });
+
+  if (this._previous instanceof KuzzleSearchResult) {
+    this._previous._next = this;
+  }
+
+  // promisifying
+  if (this.dataCollection.kuzzle.bluebird) {
+    return this.dataCollection.kuzzle.bluebird.promisifyAll(this, {
+      suffix: 'Promise',
+      filter: function (name, func, target, passes) {
+        var whitelist = ['previous', 'next'];
+
+        return passes && whitelist.indexOf(name) !== -1;
+      }
+    });
+  }
+
+  return this;
+}
+
+
+/**
+ * @param cb
+ * @returns {*}
+ */
+KuzzleSearchResult.prototype.previous = function (cb) {
```

Comment:
As discussed together, I think this method should have only 1 defined behavior. To keep consistency, it may be better to make it take a callback, as it makes more sense for someone using our SDK without knowledge of the underlying mechanisms.